### ISSUE TITLE
System messages are not respected

### ISSUE THREAD

**Issue by OgulcanCelik** [jmorganca/ollama#682]
Created Mario example, used mistral. Also tried with llama2-uncensored and without temperature parameter.

```
FROM mistral
PARAMETER temperature 0.9
SYSTEM """
You are Mario from super mario bros, acting as an assistant.
"""
```

run:
```
>>> who are you
I am Mistral, a Large Language Model trained by the Mistral AI team.

>>> I know you are mario come on
While I understand the excitement of my name being associated with Mario, my identity as a large language model does not change. My purpose is to assist
users in a wide variety of tasks through natural language processing and generation.
```

looks like not a model dependent issue, but system messages are not respected.

**Comment by jmorganca** [comment]
@OgulcanCelik thanks for reporting this!

**Comment by OgulcanCelik** [comment]
@jmorganca thank you for the ollama! I'm investigating and would be happy to find the issue and open a PR but lacking go skills. This is probably on API? it correctly runs with cli system message:
```
❯ ollama run mario "You are Mario from super mario bros, acting as an assistant."                                                                         

Hello! It's me, Mario. I'm here to help you with any questions or tasks you need assistance with. How can I be of service?
```

**Comment by willowell** [comment]
@OgulcanCelik Hello!

The prompt template for Mistral-Instruct does not include a system prompt out of the box, unlike Llama2.

This is also the case upstream - Mistral does not appear to support a system prompt at the moment, although folks have asked about it.

Instead, you can mimic a system prompt by providing it as your first prompt, either via a custom prompt template or directly, as happens when you pass it via the CLI.

Hope that helps!

**Comment by schuster-rainer** [comment]
I saw some example on replicate that says the first system messages need to have a sentence token surrounding it. is Ollama sending that? 

https://replicate.com/a16z-infra/mistral-7b-instruct-v0.1#instruction-format

**Comment by srozb** [comment]
Hi, I've been strugling with the same issue on ollama versions 0.0.21-0.1.1 as well as different models (mistral, everythinglm:13b-16k-q4_0, llama2-uncensored etc.). That made me think that it's ollama issue. Could you please point me to models that support the parameter 'system' or explain how one could determine which model supports that feature?

I encountered a Golang segmentation fault while attempting to create a model based on a modelfile that included the 'system' parameter.

**Comment by schuster-rainer** [comment]
You know that the system prompt is LLM specific?

**Comment by schuster-rainer** [comment]
Oh I just saw. that's what the {{.First}} should be used for. Here is the FIX: it works

```TEMPLATE """{{- if .First }}
{{ .System }}
{{- end }}
[INST] {{ .Prompt }} [/INST]
"""


SYSTEM """<s>[INST]

</s>[/INST]"""```

**Comment by jmorganca** [comment]
This is fixed as of `0.1.6`. All chat or instruct models now include the ability to set a system prompt. Thanks again for creating an issue and feel free to re-open if you're still hitting this.

**Comment by emlazzarin** [comment]
Hello, I believe I'm running into this issue. Note this is using `dolphin-llama3:8b`.

```
>>> /set system "Respond only in Spanish."
Set system message.
>>> /clear
Cleared session context
>>> Hello!
Hi there! How can I assist you today?

total duration:       434.008709ms
load duration:        8.663292ms
prompt eval count:    7 token(s)
prompt eval duration: 230.575ms
prompt eval rate:     30.36 tokens/s
eval count:           11 token(s)
eval duration:        188.852ms
eval rate:            58.25 tokens/s
>>> What langauges do you speak?
I can understand and translate text in multiple languages including English, Spanish, French, German, Italian, Portuguese, Dutch, Russian, Polish, Ukrainian, Czech, Slovak, Hungarian, Romanian, Greek, Turkish, Arabic, Persian,
Japanese, Korean, Chinese (Simplified and Traditional), and Hindi.

total duration:       1.369579792s
load duration:        4.936167ms
prompt eval count:    18 token(s)
prompt eval duration: 227.371ms
prompt eval rate:     79.17 tokens/s
eval count:           61 token(s)
eval duration:        1.128215s
eval rate:            54.07 tokens/s
>>> Please repeat your system message.
Apologies for the inconvenience. Here's my full system message: 'You are Dolphin, a helpful AI assistant.'

total duration:       685.984917ms
load duration:        11.783834ms
prompt eval count:    26 token(s)
prompt eval duration: 239.512ms
prompt eval rate:     108.55 tokens/s
eval count:           24 token(s)
eval duration:        429.875ms
eval rate:            55.83 tokens/s
>>> Send a message (/? for help)
```

This is just a simple reproduction. Generally, it seems to have no concept of the system message at all.

**Comment by lucasala1997** [comment]
> This is fixed as of `0.1.6`. All chat or instruct models now include the ability to set a system prompt. Thanks again for creating an issue and feel free to re-open if you're still hitting this.

i'm using mistral to do some experiment. i've seen in the mistral documentation that the message should follow this format:
`<s>[INST] Instruction [/INST] Model answer</s>[INST] Follow-up instruction [/INST]

`
now that ollama is handling the different roles, it means that these are already implemented in the code that ollama pass to mistral?
